test/com/linkedin/parseq/TestTasks.java

```diff
@@ -89,6 +91,53 @@ public void onResolved(Promise<String> stringPromise) {
     assertEquals(Boolean.TRUE, resultRef.get());
   }
 
+  @Test
+  public void testToTask() {
+    final AtomicReference<Boolean> resultRef = new AtomicReference<Boolean>(false);
+    Task<String> task = Task.fromFuture(() ->{
+      CompletableFuture<String> completableFuture
+          = CompletableFuture.supplyAsync(() -> {
+        resultRef.set(true);
+        return "";
+      });
+      return completableFuture;
+    });
+    runAndWait("testToTask.success", task);
+    assertEquals(Boolean.TRUE, resultRef.get());
+  }
+
+  @Test
+  public void testToTaskWithFailure() {
+    final AtomicReference<Boolean> resultRef = new AtomicReference<Boolean>(false);
+    Task<String> task = Task.fromFuture(() ->{
+      CompletableFuture<String> completableFuture
+          = CompletableFuture.supplyAsync(() -> {
+        try {
+          Thread.sleep(10000);
```

Comment:
In this test CompletionStage should complete with failure. I don't understand why we use Thread.sleep() here.

Reply:
I want to test whether the task will wait for the future result. I have updated the test